File: Rudim/CLI/UCI/TimeManagement.cs
using Rudim.Common;
using System;

namespace Rudim.CLI.UCI
{
    public static class TimeManagement
    {
        public static int CalculateMoveTime(int clock, int increment)
        {
            // Spread remaining clock over ~20 expected moves and add a bonus from increment
            // Cap at clock minus buffer to ensure we never timeout due to clock or network delays
            int moveTime = clock / 20 + increment / 2;
            return Math.Max(10, Math.Min(moveTime, clock - Constants.BufferTime));
        }
    }
}

File: src/uci/time_management.rs
use crate::common::constants;

pub fn calculate_move_time(clock: i32, increment: i32) -> i32 {
    let move_time = clock / 20 + increment / 2;
    std::cmp::max(
        10,
        std::cmp::min(move_time, clock - constants::BUFFER_TIME as i32),
    )
}
